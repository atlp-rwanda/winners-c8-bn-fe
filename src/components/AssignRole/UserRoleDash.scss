.Content {
    display: flex;
    background: #F5F5F5;
  
    .ContentContainer {
      flex: 6;
    }
    .titleDashboard{
        width: 95%;
        margin: 5px auto;
        display: flex;
        justify-content: space-between;
        align-items: center;

        .text{
          display: flex;
          align-items: center;
          padding: 20px;
          margin-top: 10px;
          gap: 20px;
          h3, h4,h6{
            color:#395B64;
            font-size: medium
            ;
          }
        }
    }
    .contentBlock{
      width: 95%;
      margin: 0px auto;
    }
    .labelClass{
        padding-left: 10px;
        position: absolute;
        font-style: normal;
        font-size: 15px;
        left: 220px;
        top: 250px;
        justify-content: space-between;
    }
    .inputData{
        position: absolute;
        width: 450px;
        border-color: #F9FAFB #F9FAFB #606060;
        height: 40px;
        left: 320px;
        top: 300px; 
    }
    .role{ 
        padding-left: 10px;
        position: absolute;
        font-style: normal;
        font-size: 15px;
        left: 220px;
        top: 400px;  
    }
    .OptionSelect{
        position: absolute;
        width: 450px;
        border-color: #F9FAFB #F9FAFB #606060;
        height: 40px;
        left: 320px;
        top: 450px; 
    
    }
    .buttonClass{
        position: absolute;
        display: flex;
        flex-direction: row;
        justify-content: center;
        margin-left: 300px;
        align-items: center;
        padding: 10px 60px;
        position: absolute;
        bottom: 180px;
        gap: 10px;
        right: 800px;
        background: #395B64;
        }
        .buttonText{
          color: white;
          size:100px
      }

  }
