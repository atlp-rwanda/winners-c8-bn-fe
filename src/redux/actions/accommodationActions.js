import axios from 'axios'
import { successToast, errorToast } from '../../helpers/generateToast';
import { successAlert, failureAlert} from './alertActions';
import { FETCH_ACCOMMODATIONS_LOADING, 
         FETCH_ACCOMMODATIONS_SUCCESS, 
         FETCH_ACCOMMODATIONS_FAILED,
         FETCH_SINGLE_ACCOMMODATION_LOADING,
         FETCH_SINGLE_ACCOMMODATION_SUCCESS,
         FETCH_SINGLE_ACCOMMODATION_FAILED} from "./actionTypes";
import {accommodationsUrl} from "../utils/apiUrls";
import {authHeader} from '../utils/dataSession';

export const listAccommodations = () => async dispatch => {
    dispatch({
        type: FETCH_ACCOMMODATIONS_LOADING
    });

  return await axios.get(accommodationsUrl, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_FAILED,
              payload: err.message,
              message: err
          });
          errorToast(err)
      })
}

export const detailsAccommodation = (accommodationId) => async dispatch => {
    dispatch({
        type: FETCH_SINGLE_ACCOMMODATION_LOADING
    });

  return await axios.get(accommodationsUrl+`${accommodationId}`, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_FAILED,
              payload: "failed to fetch",
              message: err
          });
      })
}
