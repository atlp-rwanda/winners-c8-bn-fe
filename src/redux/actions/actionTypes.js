export const FECTH_REQUESTS = 'FECTH_REQUESTS';
export const FECTH_REQUESTS_COMMENTS = 'FECTH_REQUESTS_COMMENTS';
export const FECTH_REQUESTS_COMMENTS_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const DELETE_REQUESTS_COMMENT = 'DELETE_REQUESTS_COMMENT';
export const DELETE_REQUESTS_COMMENT_FAILED = 'DELETE_REQUESTS_COMMENT_FAILED';

export const GOOGLE_SIGN_IN_START = 'GOOGLE_SIGN_IN_START';
export const GOOGLE_SIGN_IN_SUCCESS = 'GOOGLE_SIGN_IN_SUCCESS';
export const GOOGLE_SIGN_IN_FAIL = 'GOOGLE_SIGN_IN_FAIL';

export const FACEBOOK_SIGN_IN_START = 'FACEBOOK_SIGN_IN_START';
export const FACEBOOK_SIGN_IN_SUCCESS = 'FACEBOOK_SIGN_IN_SUCCESS';
export const FACEBOOK_SIGN_IN_FAIL = 'FACEBOOK_SIGN_IN_FAIL';

export const FETCH_ACCOMMODATIONS_LOADING = 'FETCH_ACCOMMODATIONS_LOADING';
export const FETCH_ACCOMMODATIONS_SUCCESS = 'FETCH_ACCOMMODATIONS_SUCCESS';
export const FETCH_ACCOMMODATIONS_FAILED = 'FETCH_ACCOMMODATIONS_FAILED';

export const FETCH_SINGLE_ACCOMMODATION_LOADING =
  'FETCH_SINGLE_ACCOMMODATION_LOADING';
export const FETCH_SINGLE_ACCOMMODATION_SUCCESS =
  'FETCH_SINGLE_ACCOMMODATION_SUCCESS';
export const FETCH_SINGLE_ACCOMMODATION_FAILED =
  'FETCH_SINGLE_ACCOMMODATION_FAILED';

export const FETCH_USER_LOADING = 'FETCH_USER_LOADING';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_FAILED = 'FETCH_USER_FAILED';

export const FETCH_ROLE_LOADING = 'FETCH_ROLE_LOADING';
export const FETCH_ROLE_SUCCESS = 'FECTH_ROLE_SUCCESS';
export const FETCH_ROLE_FAILED = 'FECTH_ROLE_FAILED';

export const ASSIGN_ROLE_LOADING = 'ASSIGN_ROLE_LOADING';
export const ASSIGN_ROLE_SUCCESS = 'ASSIGN_ROLE_SUCCESS';
export const ASSIGN_ROLE_FAILED = 'ASSIGN_ROLE_FAILED';

export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
export const LOGOUT_FAILED = 'LOGOUT_FAILED';

export const ACCOMMODATION_UPDATE_LOADING = 'ACCOMMODATION_UPDATE_LOADING';
export const ACCOMMODATION_UPDATE_SUCCESS = 'ACCOMMODATION_UPDATE_SUCCESS';
export const ACCOMMODATION_UPDATE_FAILURE = 'ACCOMMODATION_UPDATE_FAILURE';

export const ACCOMMODATION_DELETE_LOADING = 'ACCOMMODATION_DELETE_LOADING';
export const ACCOMMODATION_DELETE_SUCCESS = 'ACCOMMODATION_DELETE_SUCCESS';
export const ACCOMMODATION_DELETE_FAILURE = 'ACCOMMODATION_DELETE_FAILURE';

export const ACCOMMODATION_VIEW_LOADING = 'ACCOMMODATION_VIEW_LOADING';
export const ACCOMMODATION_VIEW_SUCCESS = 'ACCOMMODATION_VIEW_SUCCESS';
export const ACCOMMODATION_VIEW_FAILURE = 'ACCOMMODATION_VIEW_FAILURE';
export const APPROVEREJECT_PENDING_REQUEST = 'APPROVE_PENDING_REQUEST';
export const APPROVEREJECT_SUCCESS = 'APPROVE_REQUEST_SUCCESS';
export const APPROVEREJECT_ERROR = 'APPROVE_REQUEST_ERROR';

export const FETCH_ALL_CHATS = 'GET_CHATS';
export const FETCH_ALL_CHATS_FAILED = 'FAILED_TO_LOAD_CHATS';
export const CHAT_SUCCESS = 'CHAT_SUCCESS';
export const CHATTING_STATUS = 'CHATTING_STATUS';
