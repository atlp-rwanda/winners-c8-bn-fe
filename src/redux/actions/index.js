// this folder will contain redux's actions