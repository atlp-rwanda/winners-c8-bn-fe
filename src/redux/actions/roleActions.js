import axios from 'axios';
import { successToast, errorToast } from '../../helpers/generateToast';
import { FETCH_ROLE_FAILED, FETCH_ROLE_SUCCESS, FETCH_ROLE_LOADING, FETCH_USER_SUCCESS, FETCH_USER_FAILED, FETCH_USER_LOADING, ASSIGN_ROLE_FAILED, ASSIGN_ROLE_LOADING, ASSIGN_ROLE_SUCCESS } from './actionTypes';
import { authHeader } from '../utils/dataSession';

export const fetchUsers = () => async dispatch => {
  dispatch({
    type: FETCH_USER_LOADING,
    payload: "Fetch users is loading",
  });

return await axios.get(`https://winners-c8-bn-be-staging.herokuapp.com/api/users`, {headers: authHeader()})
  .then(res=> {
  
    dispatch({
    type: FETCH_USER_SUCCESS,
    payload: res.data,
  })
})
.catch(err =>{
  dispatch({
    type: FETCH_USER_FAILED,
    payload: "User not found",
  })
})
}

export const fetchRoles = () => async dispatch => {
      dispatch({
        type: FETCH_ROLE_LOADING,
      });
    
  return await axios.get(`https://winners-c8-bn-be-staging.herokuapp.com/api/users/roles`, {headers: authHeader()})
      .then(res=> {
        dispatch({
        type: FETCH_ROLE_SUCCESS,
        payload: res.data.roles,
      })
    })
    .catch(err => {
      dispatch({
        type: FETCH_ROLE_FAILED,
        payload: "Fetch roles failed",
      })
    })
  }
  export const assignRoles = (SelectedEmail, SelectedRole) => async dispatch => {
    dispatch({
      type: ASSIGN_ROLE_LOADING,
    });
  
return axios.patch(`https://winners-c8-bn-be-staging.herokuapp.com/api/users/assignRole`,{email: SelectedEmail, roleId: SelectedRole}, {headers: authHeader()})
    .then( async res => {
      if(res.data.error){
        await dispatch({
             type: ASSIGN_ROLE_FAILED,
             payload: res.data.error
         })
         errorToast(err.response.data.message)
     }
      // console.log("data", res.status)
      if(res.status == 200){
     await dispatch({
      type: ASSIGN_ROLE_SUCCESS,
      payload: res.data.data
    })
    successToast(res.data.message)
  }
    dispatch(fetchUsers())
  })
  .catch(err =>{
    console.log(err.response.data.message)
    dispatch({
      type: ASSIGN_ROLE_FAILED,
      payload: err,
    });
    errorToast(err.response.data.message)
  })

}

  