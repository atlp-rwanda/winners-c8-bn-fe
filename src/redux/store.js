// this file will configure the redux's store
import { createStore } from "redux";
import rootReducer from "./reducers";

let authToken = ""; 
if(window.localStorage.getItem("auth-token")) {
  authToken = window.localStorage.getItem("auth-token");
}

const initialState = {
    auth: {token: authToken},
    navbar: {currentPage: "HOME"},
};
const store = createStore(
    rootReducer,
    initialState // sets initial state
  );

export default store;
