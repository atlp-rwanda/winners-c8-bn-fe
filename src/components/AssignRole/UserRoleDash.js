import MenuItem from '@mui/material/MenuItem';
import React, { useEffect, useState } from 'react';
import "bootstrap/dist/css/bootstrap.min.css";
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { Skeleton } from "@mui/material";
import Select from '@mui/material/Select';
import { fetchRoles, fetchUsers, assignRoles } from '../../redux/actions/roleActions';
import { useDispatch, useSelector } from 'react-redux';
import "./UserRoleDash.scss"
import Navbar from "../Navbar/Navbar";
import Sidebar from "../Sidebar/Sidebar";
const UserRoleDash = () => {
        const userInfos = useSelector((state) => state.users?.users?.data)
      
      
        const roleData = useSelector((state) => state.roles?.roles)
      
        useEffect(() => {
          }, [SelectedEmail])
      
      
        const [users, setUsers] = useState()
      
      
        const [isLoading, setIsLoading] = useState(false);
      
        const [SelectedRole, setSelectedRole]  = useState(null)
      
        const [SelectedEmail, setSelectedEmail]  = useState(null)
      
        const [roles, setRoles] = useState(null)
        
        const dispatch = useDispatch()
      
      
      useEffect(() => {
      fetchUsers()(dispatch)
      }, [])
      
      useEffect(() => {
        fetchRoles()(dispatch)
        }, [])
      
      
      
        const SubmitAssignRole = (event) =>{
        event.preventDefault();
        dispatch(assignRoles(SelectedEmail, SelectedRole))
        }
        
        useEffect(() => {
          setUsers({
            firstName:userInfos?.firstName,
            lastName:userInfos?.lastName,
            email:userInfos?.email,
          })
          setIsLoading(true)
        }, [userInfos])
      
          useEffect(() => {
            setRoles({
              id:roleData?.roleId,
              role:roleData?.roleName,
            })
            setIsLoading(true)
        }, [roleData])
    return ( 
        <>
      <div className="Content" data-testid='assign-role'>
      <Sidebar />
      <div className="ContentContainer">
        <Navbar />
        <div className="titleDashboard">
          <div className="text">
           <h4>Dashboard&gt;&gt;</h4>
           <h6>Assign Role</h6>
           </div>
           <h4>Dashboard&gt;&gt;</h4>
          </div>
        
  <div className='contentBlock'>
  <p>Assign Role</p>  
  {isLoading ? (<div className="Content">
<form className='formClass'>
<label className='labelClass'>Choose an Email</label>

<Select
 className='inputData'
 data-testid="user-byEmail"
defaultValue={users?.email}
  onChange={(event) =>
    setSelectedEmail(
       event.target.value,
    )
  }
 >
     {userInfos?.map(value => {
    return (
      <MenuItem value={value.email} key={value.email}>
        {value.email}
      </MenuItem>
    )
  })}
 </Select>
<div className='role'>Select Role name</div>
<Select
 className='OptionSelect'
 data-testid="role-name-byId"
  defaultValue={roles?.role}
  onChange={(event) =>
{setSelectedRole(event.target.value)}
  }
 >
   {roleData?.map(value =>{
     return ( <MenuItem  value={value.id} key={value.id}>
       {value.roleName}
     
     </MenuItem>)
   })}
 </Select>
<button
className='buttonClass'
type="submit"
data-testid="assign-role-btn"
onClick={SubmitAssignRole}

><div className='buttonText'>assignRole</div></button> 
<ToastContainer/>
</form>
</div>):
          <Skeleton
          variant="rectangle"
          animation="wave"
          width={1100}
          height={800}
          />}
</div>
      </div>
    </div>
    </>
     );
}
 
export default UserRoleDash;