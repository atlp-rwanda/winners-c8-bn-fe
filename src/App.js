import React from 'react';
import Home from './components/Home';
import { Route, Routes } from 'react-router-dom';
import UserRegisterForm from './components/RegisterForm';
import RecoveryForm from './components/recoveryForm';
import ResetForm from './components/resetForm';
import LoginForm from './components/LoginForm';
import Userprofile from './components/UserProfile/UserProfile';
import Request from './pages/Request';
import Comments from './pages/Comments';
import Dashboard from './components/Dashboard';
import NotFound from './components/NotFound';
import UserRoleDash from "./components/AssignRole/UserRoleDash"
import { ToastContainer } from 'react-toastify';
import DashboardLayout from './components/Layouts/Dashboard';
import HomePageLayout from './components/Layouts/Home';

const App = () => {
  return (
    <div className="App">
      <ToastContainer />
      <div className="content">
        <Routes>
          <Route path="/" element={<HomePageLayout />}>
            <Route exact path="login" element={<LoginForm />} />
            <Route index element={<Home />} />
            <Route exact path="register" element={<UserRegisterForm />} />
            <Route exact path="recover" element={<RecoveryForm />} />
            <Route exact path="reset" element={<ResetForm />} />
          </Route>
          <Route path="/dashboard" element={<DashboardLayout />}>
            <Route exact index element={<Dashboard />} />
            <Route path="trips" element={<Request />} />
            <Route path="trips/trip" element={<Comments />} />
          <Route exact path='assignRole' element={<UserRoleDash />} />
            <Route exact path="userprofile" element={<Userprofile />} />
            <Route path="*" element={<NotFound />} />
          </Route>
          <Route exact path="*" element={<NotFound />} />
        </Routes>
      </div>
      {/* <SocialAuth /> */}
    </div>
  );
};

export default App;
