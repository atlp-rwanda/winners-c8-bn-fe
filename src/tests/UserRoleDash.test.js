import React from "react";
import UserRoleDash from "../components/AssignRole/UserRoleDash";
import renderer from 'react-test-renderer'
import { assignRoles } from '../../src/redux/actions/roleActions';
import { cleanup,render,screen, act, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom';
import store from '../redux/store';
import App from '../App';
import { BrowserRouter as Router } from 'react-router-dom';
import { userResponse } from "./mocks/data";
import { MemoryRouter} from 'react-router-dom';
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import MockAxiosAdapter from 'axios-mock-adapter';
import axios from "axios";
const middleware = [thunk];
 const mock = new MockAxiosAdapter(axios)

 
 describe('User role Dashboard ', () => {
beforeEach(() =>{
  mock.onGet(/user/gi).reply(200, userResponse);
})

  afterEach(() => {
    cleanup();
    window.localStorage.clear();
    mock.reset();
  });




  it('dispatches SUCCESS after assignrole', async () => {
    const userObj = {
      email: 'honore@gmail.com',
      role: 'Manager',
    };

    await store.dispatch(assignRoles(userObj));
  });




  test('should render assign role component', () => {
    render(
      <Provider store={store}>
        <Router>
          <UserRoleDash />
        </Router>
      </Provider>
    );
  const userRoleElement = screen.queryByTestId('assign-roles');
  expect(userRoleElement).toBeInTheDocument()
  expect(userRoleElement).toHaveTextContent('Assign Role');
    })

  it('The page should render  the page and show all users', async () => {
    window.localStorage.setItem('auth-token', 'testing');
    await act(() =>
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/users']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
  );

  })
  it('The page should render the page and show all roles', async () => {
    window.localStorage.setItem('auth-token', 'testing');
    await act(() =>
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/users/roles']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
  );

  })
test("<UserRoleDash /> matches snapshot", ()=>{   
  const component = renderer
		.create(
			<Provider store={store}>
				<MemoryRouter>
          <UserRoleDash />
				</MemoryRouter>
			</Provider>,
		)
		.toJSON();
    expect(component).toMatchSnapshot();
})
 })