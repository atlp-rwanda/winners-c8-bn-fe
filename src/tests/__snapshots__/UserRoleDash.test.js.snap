// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User role Dashboard  <UserRoleDash /> matches snapshot 1`] = `
<div
  className="Content"
  data-testid="assign-role"
>
  <div
    className="ContentContainer"
  >
    <div
      className="titleDashboard"
    >
      <div
        className="text"
      >
        <h4>
          Dashboard&gt;&gt;
        </h4>
        <h6
          data-testid="assign-roles"
        >
          Assign Role
        </h6>
      </div>
    </div>
    <div
      className="contentBlock"
    >
      <div
        className="Content"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-yvvig0-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                Assign role
              </span>
            </div>
          </div>
          <div
            className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <form
              className="formClass"
            >
              <div
                className="css-1lwbda4-MuiStack-root"
              >
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink={false}
                  >
                    Email
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl inputData css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                    data-testid="user-byEmail"
                    onClick={[Function]}
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="notranslate"
                      >
                        ​
                      </span>
                    </div>
                    <input
                      aria-hidden={true}
                      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      disabled={false}
                      onAnimationStart={[Function]}
                      onChange={[Function]}
                      required={false}
                      tabIndex={-1}
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink={false}
                  >
                    Role
                  </label>
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl OptionSelect css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                    data-testid="role-name-byId"
                    onClick={[Function]}
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      <span
                        className="notranslate"
                      >
                        ​
                      </span>
                    </div>
                    <input
                      aria-hidden={true}
                      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      disabled={false}
                      onAnimationStart={[Function]}
                      onChange={[Function]}
                      placeholder={
                        Array [
                          "select an email",
                        ]
                      }
                      required={false}
                      tabIndex={-1}
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
                    data-testid="assign-role-btn"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="submit"
                  >
                    assignRole
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
